src/Analyzers/CSharp/Tests/UseIndexOrRangeOperator/UseRangeOperatorTests.cs

```diff
@@ -330,25 +330,14 @@ class C
 {
     void Goo(S s)
     {
-        s.Slice([|1, s.Length - 2|]) = default;
+        s.Slice(1, s.Length - 2) = default;
     }
 }";
-            var fixedSource =
-@"
-struct S { public ref S Slice(int start, int length) => throw null; public int Length { get; } public ref S this[System.Range r] { get => throw null; } }
-class C
-{
-    void Goo(S s)
-    {
-        s[1..^1] = default;
-    }
-}";
-
             await new VerifyCS.Test
             {
                 ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
                 TestCode = source,
-                FixedCode = fixedSource,
+                FixedCode = source,
```

Comment:
this is a regression.  We should update this code.  It's fine to assign in this case because the indexer returns by-ref as well.